### ISSUE TITLE
Check for complete extraction/transformation of axioms in ncit-property-graph.ttl

### ISSUE THREAD

**Issue by mbrush** [NCI-Thesaurus/thesaurus-obo-edition#10]
On the 9-8 NCIt call we noted that the sparql queries we are using to extract knoweldge related to molecular subtyping may not be pulling everything, due to handling of 'or' in equivalent class definitions.  
@balhoff was going to look into this  further. The example from the call is below.

---------------

We compared what was pulled for "Accelerated Phase Chronic Myelogenous Leukemia, BCR-ABL1 Positive" in our sparql query (query provided at end of ticket):

Accelerated Phase Chronic Myelogenous Leukemia, BCR-ABL1   Positive | Disease_May_Have_Cytogenetic_Abnormality | i(17q)
-- | -- | --
Accelerated Phase Chronic Myelogenous Leukemia, BCR-ABL1   Positive | Disease_May_Have_Cytogenetic_Abnormality | i(19)
Accelerated Phase Chronic Myelogenous Leukemia, BCR-ABL1   Positive | Disease_May_Have_Cytogenetic_Abnormality | Loss of Chromosome Y
Accelerated Phase Chronic Myelogenous Leukemia, BCR-ABL1   Positive | Disease_May_Have_Cytogenetic_Abnormality | Monosomy 7
Accelerated Phase Chronic Myelogenous Leukemia, BCR-ABL1   Positive | Disease_May_Have_Cytogenetic_Abnormality | Trisomy   8

with what appeared on the NCIt EVS exploration tool for this term:
![image](https://user-images.githubusercontent.com/5184212/30225740-6678ef8a-9488-11e7-9368-7633477e1667.png)

and noted that all associations below the Role Group line are  missing from our data.  

--------------

**The sparql query used to pull the data from our endpoint is:**  
  
SELECT DISTINCT  ?diseaseLabel ?pLabel ?molecAbnormalityLabel 
FROM <http://purl.obolibrary.org/obo/ncit.owl>  
FROM <http://purl.obolibrary.org/obo/ncit/ncit-property-graph.ttl>  
WHERE {
?disease Disease_Has_Molecular_Abnormality: | Disease_May_Have_Molecular_Abnormality: |   Disease_Has_Cytogenetic_Abnormality: | Disease_May_Have_Cytogenetic_Abnormality: ?molecAbnormality .
?disease ?p ?molecAbnormality .
?p rdf:type owl:ObjectProperty .
?p rdfs:label ?pLabel .
?disease rdfs:label ?diseaseLabel .
?molecAbnormality rdfs:label ?molecAbnormalityLabel .
}
ORDERBY ?diseaseLabel

---------
The logical axiom for this term in the NCIt version of the owl file is:
  
Chronic Myelogenous Leukemia, BCR-ABL1 Positive' 
 and (((Disease_May_Have_Cytogenetic_Abnormality some 't(9;22)(q34.1;q11.2)')
 and (Disease_May_Have_Cytogenetic_Abnormality some 'Philadelphia Chromosome')
 and (Disease_May_Have_Molecular_Abnormality some 'p210 Fusion Protein Expression')
 and (Disease_May_Have_Molecular_Abnormality some 'p230 Fusion Protein Expression')
 and (Disease_May_Have_Molecular_Abnormality some 'p190 Fusion Protein Expression'))   
or   
 ((Disease_May_Have_Cytogenetic_Abnormality some 't(3;21)(q26;q22)')  
 and (Disease_May_Have_Molecular_Abnormality some 'EVI1-AML1 Fusion Protein Expression')))
 and (Disease_Has_Finding some Splenomegaly)
 and (Disease_Has_Finding some 'Hypercellular Bone Marrow')
 and (Disease_Has_Finding some 'Accelerated Phase of Disease')
 and (Disease_May_Have_Cytogenetic_Abnormality some 'Trisomy 8')
 and (Disease_May_Have_Cytogenetic_Abnormality some 'Monosomy 7')
 and (Disease_May_Have_Cytogenetic_Abnormality some 'i(17q)')
 and (Disease_May_Have_Cytogenetic_Abnormality some 'Loss of Chromosome Y')
 and (Disease_May_Have_Cytogenetic_Abnormality some 'i(19)')
 and (Disease_May_Have_Finding some Fatigue)
 and (Disease_May_Have_Finding some Hepatomegaly)
 and (Disease_May_Have_Finding some Thrombocytopenia)
 and (Disease_May_Have_Finding some Basophilia)
 and (Disease_May_Have_Finding some 'Weight Loss')
 and (Disease_May_Have_Finding some Leukocytosis)
 and (Disease_May_Have_Finding some Thrombocytosis)
 and (Disease_May_Have_Finding some 'Myeloblasts 10-19 Percent of Bone Marrow Nucleated Cells')
 and (Disease_May_Have_Finding some 'Myeloblasts 10-19 Percent of Peripheral Blood White Cells')
 and (Disease_May_Have_Associated_Disease some Anemia)

**Comment by balhoff** [comment]
The trouble is what the semantics of the graph edges is supposed to be. Basically it's just a flattened existential restriction relationship. So the properties within the OR aren't valid existential restrictions for the whole concept. Just to be ultra-basic, it's like saying a person is an instance of (Artist or Farmer). You can't claim the person is an instance of Artist, and you also can't claim he's an instance of Farmer. You just don't know.

It would be tricky to represent this in the flattened view. Do you think it is critical?

**Comment by mbrush** [comment]
I see your point,  but at least in this case all the atomic statements that you didn't extract from this axiom are "May_Have" assertions.  So semantically it is still valid to extract and express them as such, right?  Using your example, if a person is an instance of (Artist or Farmer), you could would be correct in making the statements that the person May_Be an Artist and that the person May_Be a Farmer.  

As an aside - do you know what  "Role Group(s)" in the NCIt explorer screen grab above means?  This section captures exactly those statements that were missing from your translation - so it may be that they are  separated here for the same reason you didn't create direct triples for them.

**Comment by fragosog** [comment]
I agree, if you take the restrictions out of the union and assert them as individual restrictions on the class, the semantics are going to be different.  I was under the impression though, that the "flattened" version did not use restrictions, but rather asserted the relations directly on the classes, as if they were individuals.  So semantics in such an owl full or rdf version wouldn't really play much of a role, the structure was meant really to support navigation not reasoning.  Hmm.

(If semantics are strict, an individual is a farmer, or is an artist, or is an artistic farmer, not sure that may_be works very well at the level of the individual - i think this is an issue with us.  Just like the use of negation to express "excludes" relations, perhaps the DL representation of "may_be" relations can be re-examined?  But this would be separate from the discussion of the OR in role groups.)

**Comment by NCIEVS** [comment]
OK, now I understand from the [wiki page](https://github.com/NCI-Thesaurus/thesaurus-obo-edition/wiki/NCIt-graph-queries) that there shouldn't be restrictions on this conversion, only the direct assertions on the class.  Perhaps I'm missing some other point, but at this time I don't think it would be an issue to parse the restrictions out of the unions in the role groups and asserting them on the classes like the other restrictions.  No?
